docs/Users/Tutorials/External_Bears.rst

```diff
@@ -0,0 +1,521 @@
+data sent to your external executable (filename, lines, settings) as a JSON
+string via ``stdin``. Afterwards the analysis takes place in your executable
+that can be written in **literally** any language. In the end you will have to
+provide the ``Results`` in a JSON string via stdout.
+
+In this tutorial we will go through 2 examples where we create a very simple
+bear. The first example will use a **compiled** language that creates a
+standalone binary, ``C++``, whilst in the second example we will take a look at
+``JS`` that needs ``node`` in order to run out of the browser.
+
+External Bear Generation Tool
+-----------------------------
+
+If you really do not want to write any Python code, there is a tool
+`here <https://gitlab.com/coala/coala-bear-management>`__,
+``coala-bears-create``, that will create the wrapper for you. We will be using
+
+::
+
+    $ coala-bears-create -ext
+
+in order to generate the wrapper for the bear.
+
+Writing a Bear in C++
+---------------------
+
+The bear that will be created with this tutorial will check whether there is any
+**coala** spelled with a capital ``C`` since that is a horrible mistake for one
+to make.
+
+1. Create a new directory and make it your current working directory.
+2. Run ``coala-bears-create`` as mentioned above in order to create the wrapper
+   for our ``C++`` bear and hit ``Enter`` for the first question. You can skip
+   this step and use the wrapper provided below.
+3. Name the bear ``coalaCheckBear`` and the executable ``coalaCheck_cpp``.
+4. Reply to the rest of the questions (hit ``Enter`` at settings). After the
+   script is finished running there should be 2 files in your current directory:
+   ``coalaCheckBear.py`` (the wrapper) and ``coalaCheckBearTest.py``.
+5. This tutorial will not focus on testing so ignore the second file for now.
+   The wrapper should look something like this (after some cleaning):
+
+::
+
+    import os
+
+    from coalib.bearlib.abstractions.ExternalBearWrap import external_bear_wrap
+
+
+    @external_bear_wrap(executable='coalaCheck_cpp',
+                        settings={})
+    class coalaCheckBear:
+        """
+        Checks for coala written with uppercase 'C'
+        """
+        LANGUAGES = {'All'}
+        REQUIREMENTS = {'coalaCheck_cpp'}
+        AUTHORS = {'Me'}
+        AUTHORS_EMAILS = {'me@mail.com'}
+        LICENSE = 'GPLv3'
+
+        @staticmethod
+        def create_arguments():
+            return ()
+
+6. Since the input will be a JSON string some kind of JSON class is needed.
+   `This one <https://github.com/nlohmann/json>`__ is a great choice because it
+   is easy to integrate and is used in this tutorial.
+7. Create ``coalaCheck.cpp`` and start by testing the input. The best thing
+   about nlohmann's JSON library is that you can parse JSON directly
+   from stdin like this:
+
+::
+
+    #include <iostream>
+
+    #include "json.hpp"
+
+    using json = nlohmann::json;
+    using namespace std;
+
+    json in;
```

Comment:
I think I didn't do it because sublime colored it differently, hinting that it is a key word.